test/templates/test_fedora.rb

```diff
@@ -36,7 +36,7 @@ def test_exclude_extension_directory
   end
 
   def test_build_requires
-    assert_match(/^# BuildRequires: rubygem\(test_development\) >= 1\.0\.0$/, @out_string)
+    assert_includes(@out_string, "\n# BuildRequires: (rubygem(test_development) >= 1.0 with rubygem(test_development) < 2 with rubygem(test_development) >= 1.0.0)\n")
```

Comment:
Just to double check, is RPM able to cope with this?

Reply:
https://fedoraproject.org/wiki/Changes/RPM-4.14#User_Experience lists `Requires: (foo >= 1.0 with foo < 2.0)` as an example. I haven't tried more 2 conditions, so that would be a good testcase. That said, there is this code in `/usr/lib/rpm/rubygems.req` (rubygems-devel-3.4.10-180.fc38.noarch)
```ruby
# Compose dependency together with its requirements in RPM rich dependency
# string.
def self.compose_dependency_string(name, requirements)
  dependency_strings = requirements.map { |requirement| name + requirement }
  dependency_string = dependency_strings.join(' with ')
  dependency_string.prepend('(').concat(')') if dependency_strings.length > 1
  dependency_string
end
```

That doesn't make any special conditions and does pretty much the same thing.